### PR TITLE
Fix `declaration-block-no-redundant-longhand-properties` with basic keywords

### LINKED ISSUE #6738: Fix `declaration-block-no-redundant-longhand-properties` autofix with basic keywords
### What steps are needed to reproduce the bug?

Hi!

Following https://github.com/stylelint/stylelint/pull/6699, `stylelint` now fixes
```css
	bottom: initial;
	left: 16px;
	right: 16px;
	top: 16px;
```
in
```css
	inset: 16px 16px initial;
```
which is incorrect, as `inset` does not support the `initial` value when position values are specified. As such, the property is ignored by browsers.

**Edit:**
I've check and it also happens with `padding-*` and `margin-*` which accept `initial` as a value but not `padding` or `margin`. Edited the title accordingly but not whole description

**Note**

:spiral_notepad:  Also, question that should maybe be a separate issue, shouldn't `*-no-redundant-*` rule not trigger on `inset` if there is a `browserlistrc` specifiying a version of browser which does not support `inset`?

### What Stylelint configuration is needed to reproduce the bug?

```json
{
	"extends": "stylelint-config-standard"
}
```

### How did you run Stylelint?

`stylelint '**/*.css'`

### Which version of Stylelint are you using?

`15.3.0`

### What did you expect to happen?

Stylelint rule should either not trigger or should fix with
```css
	inset: 16px 16px;
	bottom: initial;
```
Same for properties `margin-*`, `padding-*` and more

### What actually happened?

Stylelint fixes with unsupported values

```css
	inset: 16px 16px initial;
```

### Does the bug relate to non-standard syntax?

Nope

### Proposal to fix the bug

_No response_

### Alternative fix

In `.stylelintrc`, use 
```json
"rules": {
	"declaration-block-no-redundant-longhand-properties": [
		true,
		{
			"ignoreShorthands" : ["inset"]
		}
	]
}
```

_Also fixes and answer my question :spiral_notepad:_

### PR DESCRIPTION
**Pull request by mattxwang**
> Which issue, if any, is this issue related to?

Closes #6738.

> Is there anything in the PR that needs further explanation?

The scope of this PR is broader than autofixing, since we also shouldn't flag instances which use a basic keyword. Let me know if additional test cases / documentation are desired!

Note that we already had this feature for `inherit`; this PR just expands it to use `basicKeywords`.

If we'd like, we could also create a more "aggressive" approach that converts

```css
margin-top: inset;
margin-bottom: inset;
margin-right: inset;
margin-left: inset;
```

to

```css
margin: inset;
```

which is valid (to my understanding). The rule doesn't currently behave this way with `inherit`; but, happy to add this if we think it's helpful!

---

Spec ref for [shorthand properties not supporting CSS-wide keywords](https://w3c.github.io/csswg-drafts/css-cascade-4/#shorthand-property):

> If a [shorthand](https://w3c.github.io/csswg-drafts/css-cascade-4/#shorthand-property) is specified as one of the [CSS-wide keywords](https://www.w3.org/TR/css-values/#common-keywords) [[css-values-3]](https://w3c.github.io/csswg-drafts/css-cascade-4/#biblio-css-values-3), it sets all of its [sub-properties](https://w3c.github.io/csswg-drafts/css-cascade-4/#longhand) to that keyword, including any that are [reset-only sub-properties](https://w3c.github.io/csswg-drafts/css-cascade-4/#reset-only-sub-property). (Note that these keywords cannot be combined with other values in a single [declaration](https://w3c.github.io/csswg-drafts/css-syntax-3/#declaration), not even in a shorthand.)

### CODE REVIEW

File: lib/rules/declaration-block-no-redundant-longhand-properties/README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ a {
 }
 ```
 
-This rule will only complain if you've used the longhand equivalent of _all_ the properties that the shorthand will set and if their values are not `inherit`.
+This rule will only complain if you've used the longhand equivalent of _all_ the properties that the shorthand will set and if their values are not [CSS-wide keywords](https://www.w3.org/TR/css-values/#common-keywords) : `initial`, `inherit`, `revert`, `revert-layer`, `unset`.
```
[the remark this answers is not in the document]
Comment (mattxwang) [reply lib/rules/declaration-block-no-redundant-longhand-properties/README.md]: Seems reasonable to me!

**Review by ybiquitous — APPROVED** [review]
Thanks, LGTM 👍🏼

**Comment by pbojinov** [comment]
After upgrading to 15.4.0 I started noticing `declaration-block-no-redundant-longhand-properties` come up for styles that have a top, right, bottom, left set to 0 for position absolutes. What would be the best way to handle this inset besides disabling the next line?

```
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  // stylelint-disable-next-line declaration-block-no-redundant-longhand-properties
  left: 0;
```

**Comment by ybiquitous** [comment]
@pbojinov You mean you don't want to use the `inset` shorthand, right? If so, using a disabled comment is a way.

As another way, you can either [turn off the rule](https://stylelint.io/user-guide/configure/#rules) or [ignore only `inset`](https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties/#optional-secondary-options) via a config like this:

```json
{
  "rules": {
    "declaration-block-no-redundant-longhand-properties": null
  }
}
```

```json
{
  "rules": {
    "declaration-block-no-redundant-longhand-properties": [true, {"ignoreShorthands": ["inset"]}]
  }
}
```

**Comment by Mouvedia** [comment]
related: https://github.com/stylelint/stylelint/pull/7214#discussion_r1349613573

**Comment by mattxwang** [comment]
> related: [#7214 (comment)](https://github.com/stylelint/stylelint/pull/7214#discussion_r1349613573)

Sorry, I think I'm missing something - how is this related to the linked comment? I can submit a fix if so (since I probably introduced the bug 😓 )

**Comment by Mouvedia** [comment]
It's just a reminder so that if we tackle "CSS-wide keywords" for this rule that we take into account what has been learned from this PR. You don't have to do anything :)

**Pull request merged**